### PR TITLE
Reconfigured how 'dark-theme' stylings are applied to elements

### LINKED ISSUE #2959: Change how we turn button and text into dark theme version
Related to the recent button standardization PR that I landed.

Currently to make a dark themed button or text we simply apply `dark-theme` to the element. e.g., 

```html
<h6>regular theme heading</h6>
<h6 class="dark-theme">dark theme heading</h6>
```

If we have a section with dark background and we want to make every element in that block to be on dark theme, we will have to apply `dark-theme` class to every single element. Imagine doing that for 20 elements 🙈 ...

I want to propose that we change the way we create dark theme elements. Instead of adding `dark-theme` class to individual elements, we should have a wrapper with class `dark-theme` and have SCSS to make everything in that wrapper become their dark theme version (e.g., text should be white instead of black etc)

```html
<h6>regular theme heading</h6>

<div class="dark-theme">
  <h6>dark theme heading</h6>
  <a>dark theme anchor</a>
</div>
```

This way we can simply toggle regular mode and dark theme mode of a section by toggling `dark-theme` class on the wrapper element. 

@Pomax @youriwims @sabrinang I would love to hear your thoughts.

### PR DESCRIPTION
**Pull request by mmmavis**
# ⚠️ Warning. Do not merge until...

Check status of PR https://github.com/mozilla/foundation.mozilla.org/pull/2998 before merging this one as this PR will break footer styling implemented in that PR.

---

Closes #2959

**For design review**

Give it a quick skim through on https://foundation-mofostaging-pr-3010.herokuapp.com/en/styleguide/ to see if dark-theme elements still look as expected. (Check [production style guide](https://foundation.mozilla.org/en/style-guide/) in case you need to cross check.)sab

@sabrinang do we need to update anything on https://foundation.mozilla.org/en/docs/brand/ ?

### CODE REVIEW

**Comment by youriwims** [comment]
@mmmavis do you mind updating your initial comment to the pr that this one is dependent upon?...#3002 is the docker testing and has no footer styling.

File: source/sass/buttons.scss [hunk 1]
```diff
@@ -24,7 +24,7 @@
     box-shadow: none;
   }
 
-  &.dark-theme {
+  @at-root .dark-theme & {
```
[the remark this answers is not in the document]
Comment (mmmavis) [reply source/sass/buttons.scss]: It's a SASS syntax: https://sass-lang.com/documentation/file.SASS_REFERENCE.html#at-root

**Comment by mmmavis** [comment]
> do you mind updating your initial comment to the pr that this one is dependent upon?...#3002 is the docker testing and has no footer styling.

Oops sorry it's #2998

File: source/sass/buttons.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (mmmavis) [reply source/sass/buttons.scss]: I tried not to overuse it as it makes the styling hierarchy hard to follow sometimes (at least to me). But I think in the case it's "tidier" that we use `@at-root`. Let me know if you have other suggestion!

[the remark this answers is not in the document]
Comment (youriwims) [reply source/sass/buttons.scss]: Thanks for sharing that link! Nope—I think you're approach works great, I just wanted more info on that syntax :)

**Comment by mmmavis** [comment]
@youriwims fyi I just updated my PR (see [commit](https://github.com/mozilla/foundation.mozilla.org/pull/3010/commits/9307974b63f6b55d10b3c33528887265b92e0380)) to make sure evergreen donate section and footer section look what they want them to look.

**Comment by mmmavis** [comment]
@youriwims fyi I just updated my PR (see [commit](https://github.com/mozilla/foundation.mozilla.org/pull/3010/commits/9307974b63f6b55d10b3c33528887265b92e0380)) to make sure evergreen donate section and footer section look what we want them to look. If you could take a quick look before I merge this PR that'd be awesome!

**Comment by youriwims** [comment]
lgtm ✅
